designer/apps/core/views.py

```diff
@@ -84,3 +84,14 @@ def get(self, request):
         login(request, user)
 
         return redirect('/')
+
+def wagtail_admin_access_check(request):
+    """
+    When Wagtail tries to send a user to the cms login page, check if they are
+    already logged in. If they are redirect to an error page, showing they are
+    unauthorized to view that page. Otherwise, forward to the login page
+    """
+
+    if request.user.is_authenticated:
+        return HttpResponse('Unauthorized to access admin page', status=403)
```

Comment:
how does a user get access if they are an admin?

Reply:
The wagtail login page `/cms/login` should never be linked to directly. The only time a user will be sent to it is if Wagtail determines they don't have access to the CMS as is and need to be logged in (it will check if they are in a group with the `can_access_admin` permission).

So all this is doing is verifying that the access denial wasn't because you weren't logged in.